### PR TITLE
feat(docs): adding ask cookbook

### PR DESCRIPTION
**Pull request by TylerSehr**
Description
Add Cookbook documentation bot to Unlock docs as coordinated with Angela Steffens (TG: @wonderwomancode)

Ask Cookbook is a general purpose web3 documentation bot which queries data sources across web3 in order to generate high quality answers for developers. This implementation is focused on Unlock Protocol, and will only query secondary resources if explicitly requested.

For example, if user asks: "How do I work with Unlock Protocol contracts in Foundry?"

This will query the unlock docs and repos, and the foundry docs and repos, in order to generate answers.
This allows users to receive significantly better answers.

Issues
Sorry about this, but we weren't told about needing to create an issue, we'll happily resubmit with an issue if necessary.

### CODE REVIEW

File: .clabot [hunk 1]
```diff
@@ -63,6 +63,8 @@
     "FedericoCaruso",
     "njokuScript",
     "Calla-Ji",
+    "TylerSehr",
```
Comment (julien51) [inline .clabot:66]: Please do these changes in *another* PR.

File: docs/package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@babel/helper-get-function-arity": "7.16.7",
+    "@cookbookdev/docsbot": "*",
```
Comment (julien51) [inline docs/package.json:7]: Please use a fixed version here! It's important to avoid failing when the dependency changes

**Review by julien51 — CHANGES_REQUESTED** [review]
Can you provide me details about what this is (links?) and screenshots of the changes?

File: .clabot (hunk 1 above)
Comment (TylerSehr) [reply .clabot:66]: Opened another PR with just these changes:
https://github.com/unlock-protocol/unlock/pull/13537

**Comment by TylerSehr** [comment]
> Can you provide me details about what this is (links?) and screenshots of the changes?

Happy to!

Link to demo:
https://unlock-docs.vercel.app/ 

Screenshot 2024-03-28 at 11 07 42 AM

File: docs/src/theme/SearchBar/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export default function SearchBarWrapper(props) {
+  return (
+    <>
+      <SearchBar {...props} />
+      <BrowserOnly fallback={null}>
+        {() => {
+          const AskCookbook = require('@cookbookdev/docsbot/react')
+          return (
+            <AskCookbook
+              apiKey="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI2NWQ5OTljM2Q0NWUzNzIxM2UxZDgzMGMiLCJpYXQiOjE3MDg3NTk0OTEsImV4cCI6MjAyNDMzNTQ5MX0.2DgWnOpRQgPWfQeIOH5XRP3v1v_XDU5RSy6pqUUwwVg"
```
Comment (julien51) [inline docs/src/theme/SearchBar/index.js:14]: Can you add a comment about where to get this API key in the future please?

File: docs/src/theme/SearchBar/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+              noFastMode
+            />
+          )
+        }}
+      </BrowserOnly>
+      {/* <AskCookbook
+        apiKey="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI2NWQ5OTljM2Q0NWUzNzIxM2UxZDgzMGMiLCJpYXQiOjE3MDg3NTk0OTEsImV4cCI6MjAyNDMzNTQ5MX0.2DgWnOpRQgPWfQeIOH5XRP3v1v_XDU5RSy6pqUUwwVg"
+        noFastMode
+      /> */}
```
Comment (julien51) [inline docs/src/theme/SearchBar/index.js:23]:
```suggestion
```

**Review by julien51 — COMMENTED** [review]
Thanks! Looking good. Please clarify about how to get the API key and please resolve the conflicts...

**Comment by TylerSehr** [comment]
> Thanks! Looking good. Please clarify about how to get the API key and please resolve the conflicts...

Api key can be obtained from cms.cookbook.dev, however an invite from an admin is required. Happy to add your team for access, let's discuss on telegram.

**Comment by julien51** [comment]
> Api key can be obtained from cms.cookbook.dev, however an invite from an admin is required. Happy to add your team for access, let's discuss on telegram.

Hum. Not a huge fan of this. What happens if this API key goes down?

**Comment by TylerSehr** [comment]
Invites for access are required right now because this is in private beta. However it will be more open and self serve in the near future.

**Comment by ClockRide** [comment]
Squashed all the changes into one commit to reduce chances of merge conflicts in future

**Comment by julien51** [comment]
ok. this looks ok IMO. I am a bit troubled about the API key but I guess we can easily take that down if this creates any issue.
Also, do we have access to the questions asked by users? This would help us a lot to improve our docs.

**Comment by TylerSehr** [comment]
> ok. this looks ok IMO. I am a bit troubled about the API key but I guess we can easily take that down if this creates any issue. Also, do we have access to the questions asked by users? This would help us a lot to improve our docs.

Yes, you do. You'll have full access to all questions asked by users, all answers generated by the chatbot, ability to adjust prompt of the bot, and shortly, access to modifying the data sources used by the chatbot.

**Comment by julien51** [comment]
> Yes, you do. You'll have full access to all questions asked by users, all answers generated by the chatbot, ability to adjust prompt of the bot, and shortly, access to modifying the data sources used by the chatbot.

Cool! But how??

**Comment by TylerSehr** [comment]
> > Yes, you do. You'll have full access to all questions asked by users, all answers generated by the chatbot, ability to adjust prompt of the bot, and shortly, access to modifying the data sources used by the chatbot.
> 
> Cool! But how??

We'll send credentials to Angela Steffens and she can then invite the rest of the team. Happy to connect on Telegram for this. We just need emails to add to the management dashboard.

File: README.md [hunk 5]
```diff
@@ -30,6 +30,7 @@ Please read more about contributing in our [contributor guide](https://github.co
 Unlock uses a monorepo which includes all the services and applications we develop.
 
 ```
+
```
Comment (julien51) [inline README.md:33]:
```suggestion
```

**Review by julien51 — APPROVED** [review]

**Pull request merged**